PowerShellTools/ServiceManagement/ConnectionManager.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Diagnostics;
+using PowerShellTools.Common;
+using PowerShellTools.Common.ServiceManagement.IntelliSenseContract;
+
+namespace PowerShellTools.ServiceManagement
+{
+    /// <summary>
+    /// Manage the process and channel creation.
+    /// </summary>
+    internal sealed class ConnectionManager
+    {
+        private IPowershellIntelliSenseService _powershellServiceChannel;
+        private int _hostProcessId;
+
+        public ConnectionManager()
+        {            
+            OpenClientConnection();
+        }
+
+        /// <summary>
+        /// The service channel we need.
+        /// </summary>
+        public IPowershellIntelliSenseService PowershellServiceChannel

```

Comment:
You might want to rename "PowershellServiceChannel" into "PowershellIntelliSenseServiceChannel", coz that is going to be the naming convention debugging channel would follow.